### PR TITLE
Get and reset T_Id Sequence

### PR DESCRIPTION
**Pull request by signedav**
This is a very powerful thingy that should be handled very carefully.

#### GPKG
```
UPDATE T_KEY_OBJECT SET T_LastUniqueID = 10 WHERE T_Key = 'T_Id'
```
#### PG
```
ALTER SEQUENCE oid_madness_01.t_ili2db_seq RESTART WITH 20;
```

#### GPKG
When resetting the value and then run into conflicts.
image

Because of the unique constraints it just counts it up - means it continues to work with the tablewide t_id (I guess it takes the Geopackage Function AUTOGENERATE)...

Our t_ili_tids are of course fucked up then...

On another table of course it works but on validation of course it has troubles then because of duplicates.

#### PG

Similar but not same on postgres. When resetting the sequence, it just counts up with it, but because of unique constraint it cannot store the feature.

image

On another table of course it works but on validation of course it has troubles then because of duplicates.

### CODE REVIEW

**Comment by signedav** [comment]
Upsi. Wrong model in pg test. Will fix it...

File: modelbaker/iliwrapper/ili2dbutils.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def error_string(self):
         if self.java_version:
             return QCoreApplication.translate(
                 "ili2dbutils",
-                'Wrong java version found. Qgis Model Baker requires at least java version 8. Please <a href="{0}">install Java</a> and or <a href="{1}">configure a custom java path</a>.<br/><br/>Java Version:<br/>{2}',
+                'Wrong java version found. modelbaker requires at least java version 8. Please <a href="{0}">install Java</a> and or <a href="{1}">configure a custom java path</a>.<br/><br/>Java Version:<br/>{2}',
```
Comment (domi4484) [inline modelbaker/iliwrapper/ili2dbutils.py:350]: Is that change intentional? Should not be just "Model Baker" as it is everywhere in the GUI?

**Review by domi4484 — APPROVED** [review]

File: modelbaker/iliwrapper/ili2dbutils.py (hunk 1 above)
Comment (signedav) [reply modelbaker/iliwrapper/ili2dbutils.py:350]: I took modelbaker to make a difference between the backend "modelbaker" and the frontend "QGIS Model Baker", but since this message appears in the GUI I agree that "Model Baker" would be better. I'll change that.

File: modelbaker/dbconnector/mssql_connector.py [hunk 2]
```diff
@@ -1033,3 +1033,41 @@ def get_ili2db_settings(self):
...
+                        schema=self.schema, sequence="t_ili2db_seq", value=value
+                    )
+                )
+                self.conn.commit()
+                return True, self.tr(
+                    'Successfully reset sequence value to "{}".'
+                ).format(value)
+            except pyodbc.errors.Error as e:
+                error_message = " ".join(e.args)
+                return False, self.tr("Could not reset sequence: {}").format(
```
Comment (domi4484) [inline modelbaker/dbconnector/mssql_connector.py:1069]: Would not be better to throw an exception instead of returning false? I'm thinking more general that only this case. Or is it a design décision for the library?

**Review by domi4484 — APPROVED** [review]

File: modelbaker/dbconnector/mssql_connector.py (hunk 2 above)
Comment (signedav) [reply modelbaker/dbconnector/mssql_connector.py:1069]: Oopsi. Haven't seen this before merging. Good input, but I wanted it to make it identical to other functions. Some (e.g.) create_baskets are called for multiple baskets, where in  the frontend I 'collect' all the feedbacks and return it in a list. Kind of "5 of 10 baskets are not created" or similar.
